presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -848,6 +848,42 @@ public void testRow()
         assertQuery("SELECT cast(row(nationkey, regionkey) as row(a bigint, b bigint)) FROM nation");
     }
 
+    @Test
+    public void testDecimalSupport()
+    {
+        // Actual session is for the native query runner.
+        Session actualSession = Session.builder(getSession())
+                .setCatalog("hive")
+                .setSchema("default")
+                .setCatalogSessionProperty("hive", "parquet_pushdown_filter_enabled", "true")
+                .build();
+
+        // Expected session is for Java query runner.
+        // It does not support Parquet filter pushdown yet, so we have to explicitly disable it.
+        Session expectedSession = Session.builder(actualSession)
+                .setCatalogSessionProperty("hive", "parquet_pushdown_filter_enabled", "false")
+                .build();
+
+        // Generate temporary table name.
+        String tmpTableName = generateRandomTableName();
+
+        // Create a Parquet table with decimal types.
+        try {
+            getExpectedQueryRunner().execute(expectedSession, String.format("CREATE TABLE %s (c0 DECIMAL(15,2), c1 DECIMAL(38,2)) WITH (format = 'PARQUET')", tmpTableName));
```

Comment:
Add more tests for boundary values here.

Reply:
Updated.